### PR TITLE
SNOW-874787: Prevent future flakiness of packaging tests

### PR DESCRIPTION
**Pull request by sfc-gh-vnayak**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #NNNN

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: tests/integ/test_packaging.py [hunk 1]
```diff
@@ -52,7 +52,34 @@ def clean_up(session):
...
     session.clear_imports()
     session._runtime_version_from_requirement = None
-    yield
+
+
+@pytest.fixture(autouse=True)
+def get_available_versions_for_packages_patched(session):
+    # Save a reference to the original function
+    original_function = session._get_available_versions_for_packages
+    sentinel_version = "0.0.1"
```
Comment (sfc-gh-vnayak) [inline tests/integ/test_packaging.py:61]: The sentinel version will be returned for sktime and scikit_fuzzy, if either of those packages are present.

File: tests/integ/test_packaging.py [hunk 2]
```diff
@@ -343,6 +391,34 @@ def import_scikit_fuzzy() -> str:
     Utils.check_answer(session.sql(f"select {udf_name}()"), [Row("0.4.2")])
 
 
+@pytest.mark.skipif(
+    IS_IN_STORED_PROC,
+    reason="Subprocess calls are not allowed within stored procedures",
+)
+def test_add_requirements_unsupported_usable_by_sproc(session, resources_path):
```
Comment (sfc-gh-vnayak) [inline tests/integ/test_packaging.py:398]: This test is moved into `test_packaging.py` from `test_stored_procedure.py` file as it references the `scikit-fuzzy` package

File: tests/integ/test_packaging.py [hunk 3]
```diff
@@ -539,3 +615,20 @@ def plus_one_month(x):
...
+
+
+def test_get_available_versions_for_packages(session):
+    """
+    Assert that the utility function get_available_versions_for_packages() returns a list of versions available in Snowflake,
+    for some common packages.
+    """
+    package_table = "information_schema.packages"
+    # TODO: Use the database from fully qualified UDF name
+    if not session.get_current_database():
```
Comment (sfc-gh-aalam) [inline tests/integ/test_packaging.py:627]: do you want to take care of the todo?

Comment (sfc-gh-vnayak) [reply tests/integ/test_packaging.py:627]: Not sure what the TODO is about, it was in the codebase prior to my changes.

**Review by sfc-gh-aalam — APPROVED** [review]

**Pull request merged**